general/mem_manager.cpp

```diff
@@ -593,7 +593,7 @@ class UmpireDeviceMemorySpace : public DeviceMemorySpace,
       return CuMemcpyDtoD(dst, src, bytes);
 #endif
 #ifdef MFEM_USE_HIP
-      return HipMemcpyDtoD(dst, src, bytes);
+      return HipMemcpyDtoDAsync(dst, src, bytes);
```

Comment:
I didn't realized this was only for the `UmpireDeviceMemorySpace` when we discussed this earlier.
Do you know what would trigger the `rm.copy(dst, const_cast<void*>(src), bytes)` call?
If `Umpire` has the location knowledge, maybe it would be more appropriate to let it handle the copy.

Reply:
> I didn't realized this was only for the `UmpireDeviceMemorySpace` when we discussed this earlier.

Yea, it seemed to much of a change to do everything. I would be awesome if we could just extend the `UmpireDeviceMemorySpace` locally ourselves and do this but that's a whole other set of changes

> Do you know what would trigger the `rm.copy(dst, const_cast<void*>(src), bytes)` call?

If I'm understanding your question it looks like it is commented out right now so it would never trigger and that we used to call it but it was changed here: https://github.com/mfem/mfem/commit/c77720a203b43e944a555eff821391d7feea5421 . But maybe you are asking what Umpire _would_ do and I'm not sure of that but I will check

Reply:
updated the `HipDeviceMemorySpace` as well